### PR TITLE
upgrade cuda version on ci pipelines (training CI pipelines)

### PR DESCRIPTION
**Pull request by AdamLouly**
**Description**: 
Upgrading cuda version from 11.3 to 11.6 for all ci pipelines.
upgrading torch version to 1.12.1

### CODE REVIEW

File: orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py [hunk 1]
```diff
@@ -3746,6 +3746,8 @@ def forward(self, input1, bool_argument, int_argument, float_argument):
                 out = self.relu(out)
             return out
 
+    torch.manual_seed(0)
```
Comment (baijumeswani) [inline orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py:3749]: let's control the atol and rtol instead of setting the manual seed. I think you can try with atol=1e-5, rtol=1e-4. That should do it.

File: tools/ci_build/github/linux/docker/scripts/training/ortmodule/stage1/requirements_torch1.11.0_cu11.6/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+--pre
```
Comment (baijumeswani) [inline tools/ci_build/github/linux/docker/scripts/training/ortmodule/stage1/requirements_torch1.11.0_cu11.6/requirements.txt:1]: can you change the name of the directory of this file to reflect the correct pytorch version.

File: tools/ci_build/github/linux/run_dockerbuild.sh [hunk 3]
```diff
@@ -91,20 +91,20 @@ elif [ $BUILD_DEVICE = "gpu" ]; then
...
         fi
-        INSTALL_DEPS_EXTRA_ARGS="${INSTALL_DEPS_EXTRA_ARGS} -v 11.3"
+        INSTALL_DEPS_EXTRA_ARGS="${INSTALL_DEPS_EXTRA_ARGS} -v 11.6"
         $GET_DOCKER_IMAGE_CMD --repository "onnxruntime-$IMAGE" \
-            --docker-build-args="--build-arg BASEIMAGE=nvcr.io/nvidia/cuda:11.3.1-cudnn8-devel-${BUILD_OS} --build-arg BUILD_USER=onnxruntimedev --build-arg BUILD_UID=$(id -u) --build-arg PYTHON_VERSION=${PYTHON_VER} --build-arg INSTALL_DEPS_EXTRA_ARGS=\"${INSTALL_DEPS_EXTRA_ARGS}\" --build-arg USE_CONDA=${USE_CONDA} --network=host" \
+            --docker-build-args="--build-arg BASEIMAGE=nvcr.io/nvidia/cuda:11.6.2-cudnn8-devel-${BUILD_OS} --build-arg BUILD_USER=onnxruntimedev --build-arg BUILD_UID=$(id -u) --build-arg PYTHON_VERSION=${PYTHON_VER} --build-arg INSTALL_DEPS_EXTRA_ARGS=\"${INSTALL_DEPS_EXTRA_ARGS}\" --build-arg USE_CONDA=${USE_CONDA} --network=host" \
             --dockerfile Dockerfile.ubuntu_gpu_training --context .
 elif [[ $BUILD_DEVICE = "tensorrt"* ]]; then
-        IMAGE="$BUILD_OS-cuda11.6-cudnn8.4-tensorrt8.4"
+        IMAGE="$BUILD_OS-cuda11.6.2-cudnn8.4-tensorrt8.4"
```
Comment (baijumeswani) [inline tools/ci_build/github/linux/run_dockerbuild.sh:99]: Please don't change this. This change is not related to the training CI build.

**Comment by baijumeswani** [comment]
Please indicate in the title that this is for training CI pipelines.

**Review by baijumeswani — APPROVED** [review]

**Comment by AdamLouly** [comment]
Thanks for the review!
